subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/java/tasks/CreateDependencyInfoFile.java

```diff
@@ -0,0 +1,76 @@
+package org.shipkit.internal.gradle.java.tasks;
+
+import org.gradle.api.artifacts.*;
+import org.json.simple.JsonArray;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.notes.util.IOUtil;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Set;
+
+public class CreateDependencyInfoFile {
+
+    private static final String DESCRIPTION = "This file was generated by Shipkit Gradle plugin. See http://shipkit.org\n\n";
+
+    public void createDependencyInfoFile(CreateDependencyInfoFileTask task) {
+        JsonObject result = createJsonObject();
+        result.put("description", DESCRIPTION);
+
+        JsonArray dependencies = createJsonArray();
+
+        for (Dependency dependency: task.getConfiguration().getAllDependencies()) {
+            if (dependency instanceof ModuleDependency && !isSiblingDependency(task, dependency)) {
+                addResolvedDependency(dependencies, (ModuleDependency) dependency);
+            }
+        }
+
+        result.put("dependencies", dependencies);
+
+        IOUtil.writeFile(task.getOutputFile(), Jsoner.prettyPrint(result.toJson()));
+    }
+
+    private void addResolvedDependency(JsonArray dependencies, ModuleDependency dependency) {
+        if (!dependency.getArtifacts().isEmpty()) {
+            for (DependencyArtifact artifact : dependency.getArtifacts()) {
+                dependencies.add(getDependencyForSingleArtifact(dependency, artifact));
+            }
+        } else {
+            dependencies.add(getDependencyForSingleArtifact(dependency, null));
+        }
+    }
+
+    private boolean isSiblingDependency(CreateDependencyInfoFileTask task, Dependency dependency) {
```

Comment:
Why are we filtering out sibling dependencies? To streamline comparison?

I don't think we should be excluding dependencies on a submodule. What if we add another submodule dependency? This would be a change in pom.xml that should trigger the release.

I suggest to set version to null in that case and document why we are doing it.

```JSON
{
  "version": null,
  "submodule": true
}
```

Also, I suggest to use "submodule" or "subproject" name rather than "sibling". Sibling is a new term whereas "submodule" is something pretty well understood.

Reply:
When comparing poms there was a problem that version of submodules was always bumped so pom was never equal. But we can handle at the comparePublications level, this information should be kept here. Good point!

Reply:
I will just remove this comparison, it will simplify the task (no need for group and version as properties) and we can still figure out if it's a submodule or not in comparePublications.